Add browse folder and browse file icons

adamscott opened the pull request:
Adds the missing "browse folder" and "browse file" icons and replaces the "Browse" text button of the `ProjectDialog`, and the `FBXImporterManager` to an icon button.

New icons: ![FolderBrowse](https://github.com/godotengine/godot/assets/270928/ec0856c5-8cf3-4c3d-a528-83fab0c2d7aa) and ![FileBrowse](https://github.com/godotengine/godot/assets/270928/0920f49c-fd5a-473f-b829-0cdf84c1d955)

I opened the folder icon in Inkscape (![Folder](https://github.com/godotengine/godot/assets/270928/3755bad2-9fe9-4375-b153-f44613007d56) [`editor/icons/Folder.svg`](https://github.com/godotengine/godot/blob/81f3d43cc1ba01136795fb2059bbaa55bc514a16/editor/icons/Folder.svg)), created circles of 1px of radius, then created an outline of 1px. Subtracted the outline from the folder icon.

| Before | After |
| :---: | :---: |
|![Capture d’écran du 2024-02-25 13-15-58](https://github.com/godotengine/godot/assets/270928/eb8ad662-4aab-44a0-ba51-a5be105e3d71) | ![Capture d’écran du 2024-03-07 16-44-22](https://github.com/godotengine/godot/assets/270928/07bdc441-0852-4ea1-85d2-d2955f560a89) |
| ![Capture d’écran du 2024-02-25 17-50-30](https://github.com/godotengine/godot/assets/270928/2a8a1b81-2830-4f3a-9e24-809b0f3c3ff3) | ![Capture d’écran du 2024-03-07 16-58-15](https://github.com/godotengine/godot/assets/270928/3752900a-07d6-4d0f-b7a1-8a189269ca18) |
| ![Capture d’écran du 2024-02-26 08-59-08](https://github.com/godotengine/godot/assets/270928/cc0e5b35-108c-415e-9247-486212b598f2) | ![Capture d’écran du 2024-03-07 16-56-22](https://github.com/godotengine/godot/assets/270928/90f05b5a-dc06-407e-afe9-0dc212ae824b) |
| ![Capture d’écran du 2024-02-28 10-50-13](https://github.com/godotengine/godot/assets/270928/bc66c2ab-bc2a-4a97-b0a3-31099d168312) | ![Capture d’écran du 2024-02-28 10-48-30](https://github.com/godotengine/godot/assets/270928/1920448f-bdb4-4a15-9e6b-e187a0403044) |
| ![Capture d’écran du 2024-02-28 10-50-12](https://github.com/godotengine/godot/assets/270928/3081b8ac-f4a1-46cc-9cf9-8804f6886212) | ![Capture d’écran du 2024-02-28 10-48-26](https://github.com/godotengine/godot/assets/270928/9a00edc6-4140-48aa-b003-e991803cda85) |

Relates to #88825

# Edits
- 2024-02-26
  - Added the "browse file" icon
  - Added the browse file to the `FBXImporterManager` dialog.
  - Added the browse folder to the `EditorFileSystemImportFormatSupportQueryBlend` dialog.
- 2024-02-28
  - Changed editor settings icons depending on what is needed
- 2024-03-07
  - Added back labels to buttons.

adamscott committed:
Add browse folder and browse file icons

Mickeon reviewed (commented):
My thoughts are the same as https://github.com/godotengine/godot/pull/88825#pullrequestreview-1899721526

MewPurPur commented:
This one requires a lot more spacing as it's not grid-aligned and not multicolored.

adamscott commented:
I added more space between the dots and the icon (moved from 1px to 1.5px). 

And I added the browse file icon! ![FileBrowse](https://github.com/godotengine/godot/assets/270928/11d03fdc-da12-454d-a97b-4b213e4af14f)

adamscott commented:
I added the browse folder icon to the `EditorFileSystemImportFormatSupportQueryBlend` dialog. This should wrap up all the "browse" buttons.

MewPurPur reviewed (approved):
LGTM (literally)

AThousandShips commented:
Could you try adding it here too?
https://github.com/godotengine/godot/blob/bb6b06c81343073f10cbbd2af515cf0dac1e6549/editor/editor_resource_picker.cpp#L197-L201

adamscott commented:
> Could you try adding it here too?
> 
> https://github.com/godotengine/godot/blob/bb6b06c81343073f10cbbd2af515cf0dac1e6549/editor/editor_resource_picker.cpp#L197-L201

I don't think this PR obsoletes the load logo ![Load](https://github.com/godotengine/godot/assets/270928/7b2c75a2-e147-43ff-9447-9f38e7823141). I think that this PR means to replace the "Browse" text buttons, which don't load/open by themselves files, but only sets the path in the field.

AThousandShips commented:
True, this case here is still more a browse as it assigns a resource from a load, not opens it for editing as such

AThousandShips commented:
It's in the inspector, when you have a resource

adamscott commented:
@AThousandShips But this replaces the current resource, effectively loading it.
![image](https://github.com/godotengine/godot/assets/270928/4b4abd0b-e4a8-4fcf-bd47-e24aed73da43)

AThousandShips commented:
It loads it in the sense that it opens it in memory, but the main purpose is to replace the resource there, so I'd say it's the same as browsing for a file path, but it was just an idea :)

Calinou commented:
Should we use this icon for file properties in EditorInspector (e.g. in the Editor Settings)?

![image](https://github.com/godotengine/godot/assets/180032/b37a72c3-f67b-47b9-8df6-54f430ae560e)

adamscott commented:
> Should we use this new icon for file/directory properties in EditorInspector (e.g. in the Editor Settings)?
> 
> ![image](https://private-user-images.githubusercontent.com/180032/307957048-b37a72c3-f67b-47b9-8df6-54f430ae560e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDkxMzQ0ODYsIm5iZiI6MTcwOTEzNDE4NiwicGF0aCI6Ii8xODAwMzIvMzA3OTU3MDQ4LWIzN2E3MmMzLWY2N2ItNDdiOS04ZGY2LTU0ZjQzMGFlNTYwZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjI4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIyOFQxNTI5NDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xMmYwOTRhY2NiNWI0NjkzZmEwNWU3MjU3MDE2MjNmY2YzNmM4NzFjYzRmYzI1NDU5OTJhMGUwOGU3YzQzZmM2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.znuWokGJSet-uSQEaddsA-0WDkOhI0z48T7VAQzTycM)

@Calinou Done! See the updated images on the PR description.

adamscott commented:
@Mickeon @MewPurPur @akien-mga Following [this comment](https://github.com/godotengine/godot/pull/88825#issuecomment-1981861105), I added back text labels for the "create project" dialog, the Blender dialog and the FBX dialog. See the PR description for the updated images.

akien-mga commented on modules/gltf/editor/editor_scene_importer_blend.cpp:
```diff
@@ -515,6 +521,11 @@ bool EditorFileSystemImportFormatSupportQueryBlend::query() {
 		EditorNode::get_singleton()->get_gui_base()->add_child(browse_dialog);
+
+		// Update icons.
+		// (This is an hack because we can't rely on notifications here as we don't received them.
```
```suggestion
		// (This is a hack because we can't rely on notifications here as we don't receive them.
```

Mickeon replied:
Dare I say the parentheses are superfluous too (nitpick)

akien-mga reviewed (approved):
Looks good to me.

Mickeon reviewed (approved):
All is well what ends well

akien-mga commented:
Thanks!
